src/main/java/frc/robot/commands/BalanceCommand.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.DriveConstants;
+import frc.robot.subsystems.Drivetrain;
+
+public class BalanceCommand extends CommandBase {
+
+    private Drivetrain m_drive;
+    private final PIDController m_pid;
+
+    private double m_currentAngle, m_output;
+
+    private boolean m_usePitch;
+
+    private int m_inverted;
+ 
+    public BalanceCommand(Drivetrain drive) {
+        m_drive = drive;
+        addRequirements(drive);
+        m_pid = new PIDController(DriveConstants.kBalanceP, DriveConstants.kBalanceI, DriveConstants.kBalanceD);
+        m_pid.setTolerance(DriveConstants.kBalanceTolerance);
+    }
+
+    @Override
+    public void initialize() {
+        m_inverted = 1;
+
+        m_pid.setSetpoint(0);
+        if(Math.abs(Math.PI/2 - m_drive.getAngleHeading() % Math.PI) > Math.PI/4)  { //Determines whether to use roll or pitch
+            m_usePitch = false;
+        }
+        else  {
+            m_usePitch = true;
+        }
+        if(m_drive.getAngleHeading() > Math.PI)
+        {
+            m_inverted = -1;
+        }
+    }
+
+    @Override
+    public void execute() {
+        m_output = MathUtil.clamp(m_pid.calculate(m_currentAngle, DriveConstants.kBalanceSetpoint), -DriveConstants.kBalanceMaxOutput, DriveConstants.kBalanceMaxOutput);
+        if(m_usePitch) {
+            m_drive.driveHeading(m_output, 0, m_inverted*Math.PI/2, true);
+        }
+        else {
+            m_drive.driveHeading(m_output, 0, 0, true);
```

Comment:
heading optimization is acounted for in pitch but not yaw

Reply:
Is this optimization necessary? The driving is field relative, so positive will always be forward, even if the robot is facing backward.
One thing you guys will have to figure out is if you go on the charge station from the other side, but that's something different and only for auto as the driver in teleop would be able to get on the station.